lib/chef-vault/item.rb

```diff
@@ -340,7 +340,19 @@ def self.vault?(vault, name)
     def self.data_bag_item_type(vault, name)
       # adapted from https://github.com/opscode-cookbooks/chef-vault/blob/v1.3.0/libraries/chef_vault_item.rb
       # and https://github.com/sensu/sensu-chef/blob/2.9.0/libraries/sensu_helpers.rb
-      dbi = Chef::DataBagItem.load(vault, name)
+      begin
+        dbi = Chef::DataBagItem.load(vault, name)
+      rescue Net::HTTPServerException => http_error
+        if http_error.response.code == "404"
+          raise ChefVault::Exceptions::ItemNotFound,
+            "#{vault}/#{name} could not be found"
+        else
+          raise http_error
+        end
+      rescue Chef::Exceptions::ValidationFailed
```

Comment:
@sanga1794 we are raising Item not found exception in case of ValidationFailed ?

Reply:
just took the reference from method self.load https://github.com/chef/chef-vault/blob/6428e999cd724af42aa1da18060a2760b62393bf/lib/chef-vault/item.rb#L289